src/geode/mesh/helpers/convert_solid_mesh.cpp

```diff
@@ -161,8 +162,168 @@ namespace
+                       != polyhedron_facet_vertices.end() )
+            {
+                for( const auto v_id :
+                    geode::Indices( polyhedron_facet_vertices ) )
+                {
+                    if( polyhedron_facet_vertices.at( v_id )
+                        == ordered_vertices[4] )
+                    {
+                        ordered_vertices[7] =
+                            polyhedron_facet_vertices[( v_id + 1 ) % 4];
+                        ordered_vertices[6] =
+                            polyhedron_facet_vertices[( v_id + 2 ) % 4];
+                        break;
+                    }
+                }
+                break;
+            }
+        }
+        return ordered_vertices;
+    }
+
+    std::array< geode::index_t, 6 > order_prism_vertices(
+        const geode::index_t p, const geode::SolidMesh3D& solid )
+    {
+        std::array< geode::index_t, 6 > ordered_vertices;
+        geode::index_t already_used_facet{ 0 };
+        for( const auto f : geode::Range( 0, solid.nb_polyhedron_facets( p ) ) )
+        {
+            if( solid.nb_polyhedron_facet_vertices( { p, f } ) == 3 )
+            {
+                const auto vertices =
+                    solid.polyhedron_facet_vertices( { p, f } );
+                ordered_vertices[0] = vertices[0];
+                ordered_vertices[1] = vertices[2];
+                ordered_vertices[2] = vertices[1];
+                already_used_facet = f;
+                break;
+            }
+        }
+        for( const auto f : geode::Range( 0, solid.nb_polyhedron_facets( p ) ) )
+        {
+            if( f == already_used_facet )
+            {
+                continue;
+            }
+            const auto facet_vertices =
+                solid.polyhedron_facet_vertices( { p, f } );
+            if( absl::c_find( facet_vertices, ordered_vertices[0] )
+                    != facet_vertices.end()
+                && absl::c_find( facet_vertices, ordered_vertices[2] )
+                       != facet_vertices.end() )
+            {
+                for( const auto& v_id : geode::Indices( facet_vertices ) )
+                {
+                    if( facet_vertices.at( v_id ) == ordered_vertices[0] )
+                    {
+                        ordered_vertices[3] = facet_vertices[( v_id + 1 ) % 4];
+                        ordered_vertices[5] = facet_vertices[( v_id + 2 ) % 4];
+                        break;
+                    }
+                }
+                break;
+            }
+        }
+        for( const auto v : geode::Range( 6 ) )
+        {
+            if( ordered_vertices[1] == v || ordered_vertices[2] == v
+                || ordered_vertices[3] == v || ordered_vertices[4] == v
+                || ordered_vertices[5] == v || ordered_vertices[0] == v )
+            {
+                continue;
+            }
+            ordered_vertices[4] = v;
+            break;
```

Comment:
I don't think this does what you want (ordered_vertices should be an array of global vertex id, no ?)